R/type2.R

```diff
@@ -219,6 +224,14 @@ can_fall_back <- function(x, y) {
   has_no_proxy(x) && has_no_proxy(y)
 }
 has_no_proxy <- function(x) {
+  if (inherits(x, "vctrs:::common_class_fallback")) {
+    return(TRUE)
+  }
+
+  if (!is_null(s3_get_method(class(x)[[1]], "vec_proxy", ns = "vctrs"))) {
+    return(FALSE)
+  }
```

Comment:
If it is true that there is no `vec_proxy()` method, then isn't that enough to determine that there isn't a proxy?

i.e., do you even need to do the `vec_proxy()` call below?

Or maybe `has_no_proxy()` is not named exactly right? Maybe it is more like `proxy_is_identical_to_x()`? i.e. if there is a `vec_proxy()` method then we'd like to check that the `proxy` type exactly matches `x`.

Reply:
> If it is true that there is no vec_proxy() method, then isn't that enough to determine that there isn't a proxy?

Looks like removing the logic below doesn't break any tests, so it seems sufficient.